c/jesrequestsjson.c

```diff
@@ -0,0 +1,490 @@
+
+  DynallocDatasetName daDsn;
+  DynallocMemberName daMember;
+  memcpy(daDsn.name, dsn.value, sizeof(daDsn.name));
+  memcpy(daMember.name, memberName.value, sizeof(daMember.name));
+  DynallocDDName daDDname = {.name = JOB_ID_EMPTY};
+
+  int daRC = RC_DYNALLOC_OK, daSysRC = 0, daSysRSN = 0;
+  daRC = dynallocAllocDataset(
+      &daDsn,
+      IS_DAMEMBER_EMPTY(daMember) ? NULL : &daMember,
+      &daDDname,
+      DYNALLOC_DISP_SHR,
+      DYNALLOC_ALLOC_FLAG_NO_CONVERSION | DYNALLOC_ALLOC_FLAG_NO_MOUNT,
+      &daSysRC, &daSysRSN
+  );
+
+  if (daRC != RC_DYNALLOC_OK) {
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+    		    "error: ds alloc dsn=\'%44.44s\', member=\'%8.8s\', dd=\'%8.8s\',"
+            " rc=%d sysRC=%d, sysRSN=0x%08X (read)\n",
+            daDsn.name, daMember.name, daDDname.name, daRC, daSysRC, daSysRSN);
+    respondWithDYNALLOCError(response, daRC, daSysRC, daSysRSN,
+                             &daDsn, &daMember, "r");
+    return;
+  }
+  
+  DDName ddName;
+  memcpy(&ddName.value, &daDDname.name, sizeof(ddName.value));
+
+  respondWithSubmitDatasetInternal(response, absolutePath, &dsn, &ddName, jsonMode);
+
+  daRC = dynallocUnallocDatasetByDDName(&daDDname, DYNALLOC_UNALLOC_FLAG_NONE,
+                                        &daSysRC, &daSysRSN);
+  if (daRC != RC_DYNALLOC_OK) {
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+            "error: ds unalloc dsn=\'%44.44s\', member=\'%8.8s\', dd=\'%8.8s\',"
+            " rc=%d sysRC=%d, sysRSN=0x%08X (read)\n",
+            daDsn.name, daMember.name, daDDname.name, daRC, daSysRC, daSysRSN, "read");
+  }
+#endif /* __ZOWE_OS_ZOS */
+}
+
+int streamDatasetToINTRDR(HttpResponse* response, char *filename, int recordLength, int jsonMode){
+#ifdef __ZOWE_OS_ZOS
+  // Note: to allow processing of zero-length records set _EDC_ZERO_RECLEN=Y
+
+  int macrfParms = 0;
+  int rplParms   = 0;
+  int rc = 0;
+  char ddnameResult[DDNAME_LEN + 1] = "         ";
+  char errorBuffer[ERROR_BUF_LEN + 1];
+  unsigned int maxlrecl       = JCL_RECLEN;
+  unsigned int maxbuffer      = maxlrecl;
+
+  char *outACB = NULL;
+  RPLCommon *rpl = NULL;
+
+  rc = AllocIntReader("A", ddnameResult, errorBuffer);
+
+  outACB = openACB(ddnameResult, ACB_MODE_OUTPUT, macrfParms, 0, rplParms, maxlrecl, maxbuffer);
+      
+  if (outACB) {
+    rpl = (RPLCommon *)(outACB+RPL_COMMON_OFFSET+8);
+    if (rpl) {
+      zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG,  "rpl ok\n");
+      }
+    else {
+      zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG,  "rpl error\n");
+      }
+    }
+  else {
+    zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG,  "ACB error ok\n");
+  }
+
+  int defaultSize = DATA_STREAM_BUFFER_SIZE;
+  FILE *in;
+  if (recordLength < 1){
+    recordLength = defaultSize;
+    in = fopen(filename,"rb");
+  }
+  else {
+    in = fopen(filename,"rb, type=record");
+  }
+
+  int bufferSize = recordLength+1;
+  char buffer[bufferSize];
```

Comment:
Just use `DATA_STREAM_BUFFER_SIZE + 1`, otherwise you may end up with VLAs.

Reply:
Updated the code based on the comment